tests/AttributeTests/ResourceAttributeTest.php

```diff
@@ -281,44 +284,139 @@ public function it_can_register_resource_without_few_methods()
     }
 
     /** @test */
-    public function it_can_register_api_resource()
+    public function it_can_register_api_resource($controllerClass = ApiResource1TestController::class)
     {
-        $this->routeRegistrar->registerClass(ResourceTestApiController::class);
+        $this->routeRegistrar->registerClass($controllerClass);
 
         $this
             ->assertRegisteredRoutesCount(5)
             ->assertRouteRegistered(
-                ResourceTestApiController::class,
+                $controllerClass,
                 controllerMethod: 'index',
                 uri: 'posts',
                 name: 'posts.index'
             )
             ->assertRouteRegistered(
-                ResourceTestApiController::class,
+                $controllerClass,
                 controllerMethod: 'store',
                 httpMethods: 'post',
                 uri: 'posts',
                 name: 'posts.store'
             )
             ->assertRouteRegistered(
-                ResourceTestApiController::class,
+                $controllerClass,
                 controllerMethod: 'show',
                 uri: 'posts/{post}',
                 name: 'posts.show'
             )
             ->assertRouteRegistered(
-                ResourceTestApiController::class,
+                $controllerClass,
                 controllerMethod: 'update',
                 httpMethods: 'put',
                 uri: 'posts/{post}',
                 name: 'posts.update'
             )
             ->assertRouteRegistered(
-                ResourceTestApiController::class,
+                $controllerClass,
                 controllerMethod: 'destroy',
                 httpMethods: 'delete',
                 uri: 'posts/{post}',
                 name: 'posts.destroy'
             );
     }
+
+    /** @test */
+    public function it_can_register_api_resource_2()
+    {
+        $this->it_can_register_api_resource(ApiResource2TestController::class);
```

Comment:
This is a kinda nasty way of going about this. Please refactor to using PHPUnit's datasets.